### DISCUSSION TITLE
HW1 Part2 Q3

### DISCUSSION THREAD

**Discussion by JimmySilentG** [cooperrc/computational-mechanics#76] [Q&A]
Hello and hopefully this is the correct place to put this but I am having issue with one of the questions in HW 1. I am trying to utilize the contour function but it throws a type error saying that the array must be 2D not 3D, but it is clear that the array is 2 dimensional. I attached a screenshot of the code, although I have a feeling I'm using the function incorrectly.
![Screenshot 2022-09-02 123123](https://user-images.githubusercontent.com/105613899/188198447-433d262a-f401-4954-b182-9dcdbf74d1f0.png)

**Comment by OwenFuhr** [comment]
It might be the brackets around X and Y. Try plt.contourf(X,Y,A_99).

**Reply by cooperrc** [reply]
Yes, I think you have the brackets from the documentation, 

```python
Signature: plt.contourf(*args, data=None, **kwargs)
Docstring:
Plot filled contours.

Call signature::

    contourf([X, Y,] Z, [levels], **kwargs)
```

When the function call shows variables in brackets like that e.g. `[X, Y,]` it means the `X` and `Y` are optional, but you need both so

### these work
```python
plt.contourf(X, Y, A_99)
```
_or_
```python
plt.contourf(A_99)
```

### these don't work

```python
plt.contourf(Y, A_99)
```
_or_ 
```python
plt.contourf(X, A_99)
```
